README.md

```diff
@@ -1,189 +1,785 @@
+Blockstack is a platform for developing a new, decentralized internet, where
+users control and manage their own information. Interested developers can create
+applications for this new internet using the Blockstack platform.
+
+This repository contains a pre-release for Android developers:
+
+- the Blockstack Android SDK ([`/blockstack-sdk`](blockstack-sdk/))
+- a Android tutorial application that uses the SDK (this README tutorial)
+- tools that assist development ([`/tools`](tools/))
+
+All of the material in this is a pre-release, if you encounter an issue please
+feel free to log it [on this
+repository](https://github.com/blockstack/blockstack-android/issues).
+
+This README is written such that readers who are new to either or both
+Blockstack and Android to create a decentralized application. It contains the following sections:
+
+<!-- TOC depthFrom:1 depthTo:6 withLinks:1 updateOnSave:1 orderedList:0 -->
+
+- [Blockstack Android SDK (Pre-release)](#blockstack-android-sdk-pre-release)
+	- [Before you start](#before-you-start)
+	- [Understand the sample application flow](#understand-the-sample-application-flow)
+	- [Install the sample application dependencies and tools](#install-the-sample-application-dependencies-and-tools)
+		- [Install the Blockstack test rig](#install-the-blockstack-test-rig)
+		- [Use npm to install Yeoman and the Blockstack App Generator](#use-npm-to-install-yeoman-and-the-blockstack-app-generator)
+		- [Install Android Studio](#install-android-studio)
+	- [Build the Blockstack hello-world](#build-the-blockstack-hello-world)
+		- [Generate and launch your hello-blockstack application](#generate-and-launch-your-hello-blockstack-application)
+		- [Add a redirect end point to your application](#add-a-redirect-end-point-to-your-application)
+	- [Create the hello-android project](#create-the-hello-android-project)
+		- [Create a simple project](#create-a-simple-project)
+		- [Run the app in an emulator](#run-the-app-in-an-emulator)
+		- [Configure your application with the Blockstack SDK](#configure-your-application-with-the-blockstack-sdk)
+		- [Add a simple interface](#add-a-simple-interface)
+		- [Add session & authentication code](#add-session-authentication-code)
+		- [Run the final app in the emulator](#run-the-final-app-in-the-emulator)
+	- [Where to go next](#where-to-go-next)
+
+<!-- /TOC -->
+
+This tutorial is written for all levels from the beginner to the most
+experienced. For best results, beginners should follow the guide as written. It
+is expected that the fast or furiously brilliant will skip ahead and improvise
+on this material at will. Good speed one and all.
+
+If you prefer, you can skip working through the tutorial all together. Instead,
+you can [download the final project code](images/helloandroid.zip) and import it
+into Android Studio to review it.
+
+## Before you start
```

Comment:
Before you start contains only npm, not Android studio or yeoman.

As Android developer, I don't understand why I need npm now.

Reply:
@friedger good point. I'll sort that. Thanks for the prompt commenting.